scripts/source-maintenance.sh

```diff
@@ -33,8 +33,16 @@ KeepGoingDirective=""
 # has changed in different releases.
 # if --with-astyle /path/to/astyle is used, the check is still performed
 # and a warning is printed, but the sources are reformatted
+# --changed-only applies the formatting only to files changes since
+#    the fork point. Detection may fail, in this case it's applied to
+#    all files
+#
+# behaviouyr can be altered using environment variables:
+# ASTYLE: path to the astyle binary to use
+# MD5: path to the md5sum or similar checksum calculator to use
+
 TargetAstyleVersion="2.04"
-ASTYLE='astyle'
+ASTYLE="${ASTYLE:-astyle}"
```

Comment:
I see nothing particularly inconvenient about `--with-astyle`. Once can store source-maintenance.sh options in `.bashrc` or create a trivial wrapper script (or alias) that supplies the right options.

If `--with-astyle` is broken, it should be fixed. Adding another configuration vector to bypass an existing broken one does not sound like a step forward to me.

Reply:
It is inconvenient, in that I constantly forget to add it, and have to rerun things.

TBH this conversation is a bit frustrating. I am not forcing anybody to do something they don't like, I am adding more options so that different people with different operating styles can choose the one they like best.
All of this in an internal-use script where the last word is by automation anyway because squidadm will fix
any issue, albeit at the expense of one more commit in the tree.

The approach is fully backwards compatible, follows commonly-accepted practices and it works. Can we just move forward?

Reply:
> TBH this conversation is a bit frustrating.

Agreed.

> I am not forcing anybody to do something they don't like

You are, of course: You are essentially forcing me to review and then maintain the proposed code. As you know, the first version of this PR was buggy as hell -- Amos and I had to spend cycles to help you find problems and fix them. Squid is full of unreviewed stuff added because somebody thought it is a good idea, creating many serious long-term problems for the Project as a whole. We should not go back to that broken "if it works for X, X can merge" default!

Reply:
Let's get things on a more productive track. Intentions are the best on both sides, we have slightly different operational styles and this leads to frustration and discussions that are more heated than necessary